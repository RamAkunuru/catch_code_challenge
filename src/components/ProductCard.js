import React from 'react';

export default (props) => {
    const { name, salePrice, retailPrice, imageUrl, quantityAvailable } = props.product;

    return (
        <div className="product-card">
            <a href="#" title={name}>
                <div className="product-image">
                    <img src={imageUrl} alt={name}/>
                    <span>{quantityAvailable ? null : "SOLD OUT"}</span>
                </div>
                
                <div className="product-details">
                    <h2 className="product-name">{name}</h2>
                    <p className="product-price">{ retailPrice ? <del className="price">{retailPrice}</del> : "" }</p>
                    <p className="price sale-price">{salePrice}</p>
                </div>
            </a>
        </div>
    );
}