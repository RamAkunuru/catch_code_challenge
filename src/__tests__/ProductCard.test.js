import React from 'react';
import { shallow } from 'enzyme';
import ProductCard from '../components/ProductCard';

describe("ProductCard component", () => {
    let component;
    const product = { 
        id: "123",
        name: "abc",
        imageUrl: "https://google.com/something1",
        salePrice: 123, 
        retailPrice: 456, 
        quantityAvailable: 12
    }

    beforeEach(() => {
        component = shallow(<ProductCard product={product} />);
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    test("render matches the snapshot", () => {
        expect(component).toMatchSnapshot();
    });

    test("test if name has been rendered", () => {
        const productName = component.find('h2').text();

        expect(productName).toBe(product.name);
    });
});
