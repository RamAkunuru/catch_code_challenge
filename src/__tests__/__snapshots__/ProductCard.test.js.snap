// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard component render matches the snapshot 1`] = `
<div
  className="product-card"
>
  <a
    href="#"
    title="abc"
  >
    <div
      className="product-image"
    >
      <img
        alt="abc"
        src="https://google.com/something1"
      />
      <span />
    </div>
    <div
      className="product-details"
    >
      <h2
        className="product-name"
      >
        abc
      </h2>
      <p
        className="product-price"
      >
        <del
          className="price"
        >
          456
        </del>
      </p>
      <p
        className="price sale-price"
      >
        123
      </p>
    </div>
  </a>
</div>
`;
